core/services/feeds/service.go

```diff
@@ -43,3 +86,77 @@ func (s *service) GetManager(id int32) (*FeedsManager, error) {
 func (s *service) CountManagers() (int64, error) {
 	return s.orm.CountManagers()
 }
+
+func (s *service) CreateJobProposal(jp *JobProposal) (uint, error) {
+	return s.orm.CreateJobProposal(context.Background(), jp)
+}
+
+func (s *service) Start() error {
+	// Fetch the server's public key
+	keys, err := s.ks.ListCSAKeys()
+	if err != nil {
+		return err
+	}
+	if len(keys) < 1 {
+		return errors.New("CSA key does not exist")
+	}
+
+	privkey, err := s.ks.Unsafe_GetUnlockedPrivateKey(keys[0].PublicKey)
+	if err != nil {
+		return err
+	}
+
+	// We only support a single feeds manager right now
+	mgrs, err := s.ListManagers()
+	if err != nil {
+		return err
+	}
+	if len(mgrs) < 1 {
+		return errors.New("no feeds managers registered")
+	}
+
+	mgr := mgrs[0]
+
+	go func() {
+		err := s.connect(mgr.URI, privkey, mgr.PublicKey, mgr.ID)
+		if err != nil {
+			logger.Infof("Error connecting to Feeds Manager server: %v", err)
+		}
+	}()
+
+	return nil
+}
+
+// Connect attempts to establish a connection to the Feeds Manager.
+//
+// In the future we will connect to multiple Feeds Managers
+func (s *service) connect(uri string, privkey []byte, pubkey []byte, feedsManagerID int32) error {
+	conn, err := wsrpc.Dial(uri,
+		wsrpc.WithTransportCreds(privkey, ed25519.PublicKey(pubkey)),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Initialize a new wsrpc client to make RPC calls
+	s.mu.Lock()
```

Comment:
If we only start once, what is the point of the mutex here?

Reply:
I've put this in for the future because we will need to manage a list of connections when we decide to support multiple FMS.